src/main/java/com/itproject/petshome/controller/AdminController.java

```diff
@@ -45,14 +47,23 @@ public PetDTO deletePet(PetDTO petDTO) {
 
     @Operation(summary = "view adoption applications")
     @GetMapping("/adoptionApplication")
-    public List<AdoptionApplicationDTO> viewAdoptionApplications(@RequestParam Adopted adopted){
-        return this.userService.viewAdoptionApplications(adopted);
+    public List<AdoptionApplicationDTO> viewAdoptionApplications(@RequestParam ApplicationStatus status){
+        return this.userService.viewAdoptionApplications(status);
     }
 
     @Operation(summary = "approve or reject adoption application")
-    @PutMapping("/adoptionApplication")
-    public AdoptionApplicationDTO updateAdoptionApplication() {
-        return this.adminService.updateAdoptionApplication();
+    @PutMapping("/adoptionApplication/{id}")
+    public AdoptionApplicationDTO updateAdoptionApplication
+            (@RequestBody @Valid AdoptionApplicationDTO adoptionApplicationDTO,
+            @PathVariable("id") Long adoptionApplicationId) {
+        return this.adminService.updateAdoptionApplication
+                (adoptionApplicationDTO, adoptionApplicationId);
+    }
+
+    @Operation(summary = "view volunteer applications")
+    @GetMapping("/volunteerApplication/{id}")
+    public List<VolunteerApplicationDTO> viewVolunteerApplications(@RequestParam ApplicationStatus status) {
```

Comment:
should also have path variable here

Reply:
Would it be better to return a list of VolunteerApplicationDTO and remove path variable?
Or should it be returning an instance of VolunteerApplicationDTO and add path variable for that specific VolunteerApplicationDTO?

Reply:
I guess for admin, they should get a list of applications as well as the specific application